src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -445,34 +448,44 @@ private static bool TryParseExactN(ReadOnlySpan<char> guidString, ref GuidResult
                 return false;
             }
 
-            ref Guid g = ref result._parsedGuid;
-
-            uint uintTmp;
-            if (uint.TryParse(guidString.Slice(0, 8), NumberStyles.AllowHexSpecifier, null, out Unsafe.As<int, uint>(ref g._a)) && // _a
-                uint.TryParse(guidString.Slice(8, 8), NumberStyles.AllowHexSpecifier, null, out uintTmp)) // _b, _c
+            unchecked
             {
-                g._b = (short)(uintTmp >> 16);
-                g._c = (short)uintTmp;
+                ref Guid g = ref result._parsedGuid;
 
-                if (uint.TryParse(guidString.Slice(16, 8), NumberStyles.AllowHexSpecifier, null, out uintTmp)) // _d, _e, _f, _g
-                {
-                    g._d = (byte)(uintTmp >> 24);
-                    g._e = (byte)(uintTmp >> 16);
-                    g._f = (byte)(uintTmp >> 8);
-                    g._g = (byte)uintTmp;
+                uint uintTmp = 0;
+                uint uintTmp2 = 0;
 
-                    if (uint.TryParse(guidString.Slice(24, 8), NumberStyles.AllowHexSpecifier, null, out uintTmp)) // _h, _i, _j, _k
-                    {
-                        g._h = (byte)(uintTmp >> 24);
-                        g._i = (byte)(uintTmp >> 16);
-                        g._j = (byte)(uintTmp >> 8);
-                        g._k = (byte)uintTmp;
+                if (!TryParseHexN(guidString.Slice(0, 4), ref uintTmp) ||
+                    !TryParseHexN(guidString.Slice(4, 4), ref uintTmp2))
+                    goto FalseExit;
+                g._a = (int)((uintTmp << 16) + uintTmp2);
 
-                        return true;
-                    }
-                }
+                if (!TryParseHexN(guidString.Slice(8, 4), ref uintTmp) ||
+                    !TryParseHexN(guidString.Slice(12, 4), ref uintTmp2))
+                    goto FalseExit;
+
+                g._b = (short)uintTmp;
+                g._c = (short)uintTmp2;
+
+                if (!TryParseHexN(guidString.Slice(16, 4), ref uintTmp) ||
```

Comment:
There is still quite a bit of manual code unrolling. It would be better to prefer compact code instead. I have tried to prototype what it may look like for `n` format: https://github.com/jkotas/coreclr/commit/dfbac223ad9f2a3bff63db156d91928d3f2e2710

It is much smaller code, and it is runs faster on the microbenchmark than what is here. 

TryParseD can use similar treatment.